packages/Popover/src/Popover.js

```diff
@@ -243,15 +247,12 @@ class Popover extends React.Component {
     }
   }, throttleDelay);
 
-  handleKeyUp = event => {
-    if (event.key === "Escape") {
+  handleKeyDown = event => {
+    if (event.key === "Escape" && this.isOpen() && this.$trigger) {
```

Comment:
Perhaps could just put put `this.isOpen() && this.$trigger` as a boolean variable at the top of the function

Reply:
Hey, this is _your_ PR, man! 😜

Reply:
Ha, OUR pr now. I thinks it fine to leave

Reply:
I normally prefer having most of the code where people can find it before splitting it into variable and files, just move thing out or abstract to a variable when is clear there is duplication or abstraction is need it.